demos/sample_azure_iot/sample_azure_iot.c

```diff
@@ -0,0 +1,625 @@
+{
+    ( void ) pvContext;
+
+    LogInfo( ( "Cloud message payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               pxMessage->pvMessagePayload ) );
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Direct method message callback handler
+ */
+static void prvHandleDirectMethod( AzureIoTHubClientMethodRequest_t * pxMessage,
+                                   void * pvContext )
+{
+    LogInfo( ( "Method payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               pxMessage->pvMessagePayload ) );
+
+    AzureIoTHubClient_t * xHandle = ( AzureIoTHubClient_t * ) pvContext;
+
+    if( AzureIoTHubClient_SendMethodResponse( xHandle, pxMessage, 200,
+                                              NULL, 0 ) != eAzureIoTHubClientSuccess )
+    {
+        LogInfo( ( "Error sending method response\r\n" ) );
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Twin mesage callback handler
+ */
+static void prvHandleDeviceTwinMessage( AzureIoTHubClientTwinResponse_t * pxMessage,
+                                        void * pvContext )
+{
+    ( void ) pvContext;
+
+    switch( pxMessage->xMessageType )
+    {
+        case eAzureIoTHubTwinGetMessage:
+            LogInfo( ( "Device twin document GET received" ) );
+            break;
+
+        case eAzureIoTHubTwinReportedResponseMessage:
+            LogInfo( ( "Device twin reported property response received" ) );
+            break;
+
+        case eAzureIoTHubTwinDesiredPropertyMessage:
+            LogInfo( ( "Device twin desired property received" ) );
+            break;
+
+        default:
+            LogError( ( "Unknown twin message" ) );
+    }
+
+    LogInfo( ( "Twin document payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               pxMessage->pvMessagePayload ) );
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Setup transport credentials.
+ */
+static uint32_t prvSetupNetworkCredentials( NetworkCredentials_t * pxNetworkCredentials )
+{
+    pxNetworkCredentials->disableSni = democonfigDISABLE_SNI;
+    /* Set the credentials for establishing a TLS connection. */
+    pxNetworkCredentials->pRootCa = ( const unsigned char * ) democonfigROOT_CA_PEM;
+    pxNetworkCredentials->rootCaSize = sizeof( democonfigROOT_CA_PEM );
+    #ifdef democonfigCLIENT_CERTIFICATE_PEM
+        pxNetworkCredentials->pClientCert = ( const unsigned char * ) democonfigCLIENT_CERTIFICATE_PEM;
+        pxNetworkCredentials->clientCertSize = sizeof( democonfigCLIENT_CERTIFICATE_PEM );
+        pxNetworkCredentials->pPrivateKey = ( const unsigned char * ) democonfigCLIENT_PRIVATE_KEY_PEM;
+        pxNetworkCredentials->privateKeySize = sizeof( democonfigCLIENT_PRIVATE_KEY_PEM );
+    #endif
+
+    return 0;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Azure IoT demo task.
```

Comment:
What is a "demo task"?  I take it this basically also acts as the (de facto) main()?  Feel this could use another line or two about this is the main interaction point or some such?